cscs-checks/prgenv/environ_check.py

```diff
@@ -58,58 +58,62 @@ def set_sanity(self):
         self.sanity_patterns = sn.assert_found(module_patt, self.stderr)
 
 
-class CrayVariablesCheckMixin:
-    def __init__(self, module_name):
+class CrayVariablesCheckBase(rfm.RunOnlyRegressionTest):
```

Comment:
I suggest defining the parameter as abstract in the base class, so that nobody can accidentally create a test out of it without defining the parameters.

```suggestion
class CrayVariablesCheck(rfm.RunOnlyRegressionTest):
    cray_modules = parameter()
```